examples/medplum-patient-intake-demo/src/bots/core/intake-utils.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  createReference,
+  getExtension,
+  getReferenceString,
+  HTTP_HL7_ORG,
+  HTTP_TERMINOLOGY_HL7_ORG,
+  LOINC,
+  MedplumClient,
+} from '@medplum/core';
+import { CodeableConcept, Coding, Observation, Patient, QuestionnaireResponseItemAnswer } from '@medplum/fhirtypes';
+
+export async function upsertObservation(
+  medplum: MedplumClient,
+  patient: Patient,
+  code: CodeableConcept,
+  category: CodeableConcept,
+  valueCoding: QuestionnaireResponseItemAnswer | undefined
+): Promise<void> {
+  if (!valueCoding) {
+    return;
+  }
+
+  const observation = createObservation(patient, code, category, valueCoding);
+
+  await medplum.upsertResource(observation, {
+    code: LOINC + `|${code.coding?.[0].code}`,
+    subject: getReferenceString(patient),
+  });
+}
+
+function createObservation(
+  patient: Patient,
+  coding: CodeableConcept,
+  category: CodeableConcept,
+  valueCoding: Coding
+): Observation {
+  return {
+    resourceType: 'Observation',
+    status: 'final',
+    subject: createReference(patient),
+    code: coding,
+    category: [category],
+    valueCodeableConcept: {
+      coding: [valueCoding],
+    },
+  };
+}
+
+type ValueXAttribute = 'valueCoding' | 'valueBoolean';
+
+export function setExtension(
+  patient: Patient,
+  url: string,
+  valueXAttribute: ValueXAttribute,
+  answer: QuestionnaireResponseItemAnswer | undefined
+): void {
+  let value = answer?.[valueXAttribute];
+
+  if (valueXAttribute === 'valueBoolean') {
+    value = !!value;
+  }
+
+  if (value === undefined) {
+    return;
+  }
+
+  const extension = getExtension(patient, url);
+
+  if (extension) {
+    (extension as any)[valueXAttribute] = value;
```

Comment:
Not fully happy with this but that's the only way I could keep this function generic to all types. LMK if you prefer to have separated functions to Boolean and Coding

Reply:
Since `answer` will be either `{valueCoding: ...}` or `{valueBoolean: ...}`, could you copy the answer directly into the extension rather than, than extracting `value` and then adding it back using ` (extension as any)[valueXAttribute] = value;`